src/test/java/core/basesyntax/RegistrationServiceTests.java

```diff
@@ -0,0 +1,147 @@
+    public void register_existingLogin_throwsException() {
+        User newUser = new User();
+        newUser.setLogin("Bob679");
+        newUser.setAge(22);
+        newUser.setPassword("1230988686");
+        InvalidDataException exception = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(newUser);
+        });
+        assertEquals("The user with the same login already exists", exception.getMessage());
+    }
+
+    @Test
+    public void register_nullLogin_throwsException() {
+        User newUser = new User();
+        newUser.setLogin(null);
+        newUser.setAge(35);
+        newUser.setPassword("853895378");
+        InvalidDataException exception = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(newUser);
+        });
+        assertEquals("The login cannot be null", exception.getMessage());
+    }
+
+    @Test
+    public void register_emptyLogin_throwsException() {
+        User newUser = new User();
+        newUser.setLogin("");
+        newUser.setAge(35);
+        newUser.setPassword("59494866489640");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(newUser);
+        });
+    }
+
+    @Test
+    public void register_incorrectPassword_throwsException() {
+        User newUser = new User();
+        newUser.setLogin("Alice12");
+        newUser.setAge(44);
+        newUser.setPassword("12345");
+        InvalidDataException exception = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(newUser);
+        });
+        assertEquals("The length of password is less than 6 characters", exception.getMessage());
+    }
+
+    @Test
+    public void register_nullPassword_throwsException() {
+        User newUser = new User();
+        newUser.setLogin("Olha906784");
+        newUser.setAge(35);
+        newUser.setPassword(null);
+        InvalidDataException exception = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(newUser);
+        });
+        assertEquals("The password cannot be null", exception.getMessage());
+    }
+
+    @Test
+    public void register_emptyPassword_throwsException() {
+        User newUser = new User();
+        newUser.setLogin("Anton69504");
+        newUser.setAge(35);
+        newUser.setPassword("");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(newUser);
+        });
+    }
+
+    @Test
+    public void register_incorrectAge_throwsException() {
+        User newUser = new User();
+        newUser.setLogin("Mary67890");
+        newUser.setAge(17);
+        newUser.setPassword("098765543211111");
+        InvalidDataException exception = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(newUser);
+        });
+        assertEquals("The age is less than 18 years", exception.getMessage());
```

Comment:
The test method `register_incorrectAge_throwsException` should be more descriptive, like `register_ageUnder18_throwsException` to reflect the tested state more accurately.